test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +43,286 @@ class SubmissionServiceSpec
     with MockSoleTraderMatchingService
     with MockBusinessVerificationService
     with MockCreateTrnService
+    with MockRegistrationOrchestrationService
     with FeatureSwitching {
 
   object TestService extends SubmissionService(
-    mockJourneyService,
     mockSoleTraderMatchingService,
     mockSoleTraderIdentificationService,
     mockBusinessVerificationService,
-    mockCreateTrnService)
+    mockCreateTrnService,
+    mockRegistrationOrchestrationService
+  )
 
   implicit val hc: HeaderCarrier = HeaderCarrier()
 
+  "submit: sole trader journey with businessVerificationCheck = false" when {
+
+      "the user has a sautr" should {
+        s"register Without Business Verification" in {
+
+          val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+          mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+          mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Right(true)))
+          mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+          mockRegisterWithoutBusinessVerification(testJourneyId, testIndividualDetails.optNino, testIndividualDetails.optSautr.get)(Future.successful(Registered("someId")))
+
+          val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+          result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl + s"?journeyId=$testJourneyId")
+
+        }
+      }
+
+    "the user has no sautr" should {
+      s"not be registered"  in {
+        val matchingResult = Random.nextBoolean()
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoSautr)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetailsNoSautr, journeyConfigWithoutBV)(Future.successful(Right(matchingResult)))
+        mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "the user has no Nino and enable(EnableNoNinoJourney)" should {
+      s"not register but create a trn"  in  {
+        val matchingResult = Random.nextBoolean()
```

Comment:
What is this? Would it be it be clearer to have the mockMatchSoleTraderDetailsNoNino (which uses it 5 lines below) just return true or false. Reading it I'm unsure what the result of the matching is

Reply:
if the matching is not mention I meant is does not cover a role in the spec ... that's why I used a random boolean for matching result .... I wanted to cover 2 scenarios with 1 test .....